### PR TITLE
fix(html): do not update total time when filtering tests

### LINKED ISSUE #27758: [BUG]: HTML report incorrect total time
HTML report used to show the total execution time for all tests to finish, but now it seems like it calculates the combined total time it took for all tests to run instead.

Pictures will probably tell more.
image
image
image

Previous playwright version: 1.34.3
Current playwright version: 1.39.0

You can reproduce that by running a number of tests in parallel.

### PR DESCRIPTION
**Pull request by yury-s**
image

Fixes #27758

### CODE REVIEW

File: packages/html-reporter/src/testFilesView.tsx [hunk 1]
```diff
@@ -45,12 +46,12 @@ export const TestFilesView: React.FC<{
...
-      {!filter.empty() && <div data-testid="filtered-tests-count" style={{ color: 'var(--color-fg-subtle)', padding: '0 10px' }}>Filtered: {filteredStats.total}</div>}
+      {!filter.empty() && <div data-testid="filtered-tests-count" style={{ color: 'var(--color-fg-subtle)', padding: '0 10px' }}>Filtered: {filteredStats.total} {filteredStats.total && ('(' + msToString(filteredStats.duration) + ')')}</div>}
       <div style={{ flex: 'auto' }}></div>
       <div data-testid="overall-time" style={{ color: 'var(--color-fg-subtle)', marginRight: '10px' }}>{report ? new Date(report.startTime).toLocaleString() : ''}</div>
-      <div data-testid="overall-duration" style={{ color: 'var(--color-fg-subtle)' }}>Total time: {msToString(filteredStats.duration)}</div>
+      <div data-testid="overall-duration" style={{ color: 'var(--color-fg-subtle)' }}>Total time: {msToString(duration)}</div>
     </div>
-    {report && report.errors.length && <AutoChip header='Errors' dataTestId='report-errors'>
+    {report && !!report.errors.length && <AutoChip header='Errors' dataTestId='report-errors'>
```
Comment (yury-s) [inline packages/html-reporter/src/testFilesView.tsx:54]: It's a drive-by. If the expression evaluates to number 0, it is added as text on the page.

Comment (dgozman) [reply packages/html-reporter/src/testFilesView.tsx:54]: Ouch!!!

File: packages/html-reporter/src/testFilesView.tsx [hunk 2]
```diff
@@ -45,12 +46,12 @@ export const TestFilesView: React.FC<{
   return <>
     <div className='mt-2 mx-1' style={{ display: 'flex' }}>
       {projectNames.length === 1 && !!projectNames[0] && <div data-testid="project-name" style={{ color: 'var(--color-fg-subtle)' }}>Project: {projectNames[0]}</div>}
-      {!filter.empty() && <div data-testid="filtered-tests-count" style={{ color: 'var(--color-fg-subtle)', padding: '0 10px' }}>Filtered: {filteredStats.total}</div>}
+      {!filter.empty() && <div data-testid="filtered-tests-count" style={{ color: 'var(--color-fg-subtle)', padding: '0 10px' }}>Filtered: {filteredStats.total} {filteredStats.total && ('(' + msToString(filteredStats.duration) + ')')}</div>}
```
Comment (dgozman) [inline packages/html-reporter/src/testFilesView.tsx:49]: Same issue here?

```suggestion
      {!filter.empty() && <div data-testid="filtered-tests-count" style={{ color: 'var(--color-fg-subtle)', padding: '0 10px' }}>Filtered: {filteredStats.total} {!!filteredStats.total && ('(' + msToString(filteredStats.duration) + ')')}</div>}
```

File: packages/html-reporter/src/testFilesView.tsx [hunk 3]
```diff
@@ -29,8 +29,9 @@ export const TestFilesView: React.FC<{
   setExpandedFiles: (value: Map<string, boolean>) => void,
   filter: Filter,
   filteredStats: FilteredStats,
+  duration: number,
```
Comment (dgozman) [inline packages/html-reporter/src/testFilesView.tsx:32]: It seems like this is always equal to `report?.duration ?? 0`

File: tests/playwright-test/reporter-html.spec.ts [hunk 4]
```diff
@@ -1359,7 +1359,8 @@ for (const useIntermediateMergeReport of [false, true] as const) {
         {
           const testDuration = await page.getByTestId('test-duration').textContent();
           const totalDuration = await page.getByTestId('overall-duration').textContent();
-          expect(totalDuration).toBe('Total time: ' + testDuration);
+          await expect(page.getByTestId('overall-duration')).toHaveText(`Total time: ${msToString(result.report.stats.duration)}`);
+          expect(totalDuration).not.toBe('Total time: ' + testDuration);
```
Comment (dgozman) [inline tests/playwright-test/reporter-html.spec.ts:1363]: I'd drop this assert - it does not bring any value :)

File: tests/playwright-test/reporter-html.spec.ts [hunk 5]
```diff
@@ -1379,7 +1380,8 @@ for (const useIntermediateMergeReport of [false, true] as const) {
...
```
Comment (dgozman) [inline tests/playwright-test/reporter-html.spec.ts:1384]: same here

**Review by dgozman — APPROVED** [review]

File: packages/html-reporter/src/testFilesView.tsx (hunk 3 above)
Comment (yury-s) [reply packages/html-reporter/src/testFilesView.tsx:32]: Indeed.

**Pull request merged**